impl/src/main/kotlin/nl/nuts/consent/bridge/api/StatusApi.kt

```diff
@@ -28,18 +32,44 @@ import org.springframework.web.bind.annotation.RequestMethod
  * Simple REST service for determining if the bridge is running.
  */
 @Controller
-@RequestMapping("\${api.base-path:}")
+@RequestMapping("\${api.base-path:}/status")
 class StatusApi {
+    @Autowired
+    lateinit var healthAggregator: OrderedHealthAggregator
+
+    @Autowired
+    lateinit var healthIndicatorRegistry: HealthIndicatorRegistry
+
     /**
-     * REST service that always returns a 200 OK with an "OK" body.
+     * Http service that always returns a 200 OK with an "OK" body.
      * Can be used for checking if the bridge is up and running.
      */
     @RequestMapping(
-            value = ["/status"],
+            value = ["/"],
             produces = ["text/plain"],
             method = [RequestMethod.GET])
     fun getStatus() : ResponseEntity<String> {
         // NOP == OK
         return ResponseEntity("OK", HttpStatus.OK)
     }
+
+    /**
+     * Http service that always returns a 200 OK with an overview of component health.
+     * Can be used for debugging and manual checks.
+     */
+    @RequestMapping(
+            value = ["/diagnostics"],
+            produces = ["text/plain"],
+            method = [RequestMethod.GET])
+    fun diagnostics() : ResponseEntity<String> {
+        val h = healthAggregator.aggregate(healthIndicatorRegistry.all.map { it.key to it.value.health() }.toMap())
+        val buf = StringBuffer()
```

Comment:
In general you'd want to use StringBuilder (rather than StringBuffer), since StringBuffer is thread-safe (StringBuilder isn't) so it has a tiny overhead due to synchronisation.

Micro optimisation though.

Reply:
changed to builder and massive chain